### PR TITLE
Uses `CliprdrBackend: AsAny`

### PR DESCRIPTION
**Pull request by ibeckermayer**
Corresponds with https://github.com/Devolutions/IronRDP/pull/219/

### CODE REVIEW

File: lib/srv/desktop/rdp/rdpclient/src/client.rs [hunk 1]
```diff
@@ -318,7 +308,18 @@ impl Client {
...
+                                let mut x224_processor = Self::x224_lock(&x224_processor)?;
+                                let cliprdr = x224_processor
+                                    .get_svc_processor_mut::<Cliprdr>()
+                                    .ok_or(ClientError::InternalError)?
+                                    .backend
+                                    .as_any_mut()
+                                    .downcast_mut::<TeleportCliprdrBackend>()
+                                    .ok_or(ClientError::InternalError)?;
+                                cliprdr.set_clipboard_data(Some(data.clone()));
+                            } // unlock x224_processor
```
Comment (avatus) [inline lib/srv/desktop/rdp/rdpclient/src/client.rs:321]: I'm unsure what this comment corresponds to. The block before it? it looks like the block before locks the `clipboard_data` (shooting in the dark here, have 0% rust knowledge)

File: lib/srv/desktop/rdp/rdpclient/src/cliprdr.rs [hunk 2]
```diff
@@ -86,42 +87,47 @@ impl CliprdrBackend for TeleportCliprdrBackend {
...
+            })
+            .map(ClipboardFormat::id);
+
+        if let Some(format) = format {
             self.send("remote_copy", move |c| c.initiate_paste(format))
+        } else {
+            debug!(
+                "data was copied on the remote desktop, but no supported formats were found: {:?}",
+                available_formats
+            );
```
Comment (avatus) [inline lib/srv/desktop/rdp/rdpclient/src/cliprdr.rs:105]: What does the user see if they copied something with no supported formats. Is it this debug message?

**Review by avatus — COMMENTED** [review]
I'm not sure I'm qualified to approve or not. Looks cool tho.

Also, I always receive the same lint message as seen at the bottom of these file changes
```
Cannot find module 'teleport/ironrdp/pkg' or its corresponding type declarations.
```

Is there something I can do locally to make it go away? (its ok if not)

**Comment by probakowski** [comment]
Thanks @ibeckermayer, it looks much better, I'll merge it to my branch and work on it there

**Pull request merged**